### ISSUE TITLE
Using --tmpfs as docker parameter.

### ISSUE THREAD

**Issue by gsf-greece** [fabric8io/docker-maven-plugin#455]
I am using the official mysql image with docker-maven-plugin. I have created a directory in my host filesystem and mounted it with a tmpfs filesystem. Then I'm running my integration tests on a container from that image. Is it somehow possible to use the --tmpfs docker run parameter described  here https://docs.docker.com/engine/reference/commandline/run/ so I can skip most of these steps?

I'm only using docker-maven-plugin for 1 day so this might be something very easy that I've not found in the documentation yet. Thanks

**Comment by rhuss** [comment]
Currently I dont see any API hook for creating a tempfs. It seems to be a pure client functionality. However, as soon as we found out how this work technically I'm open to include this into the plugin, too.

**Comment by gsf-greece** [comment]
Hi again, I opened a ticket in docker project and asked them about it, the response I got is that it is already implemented [here](https://github.com/docker/docker/blob/9f0593990f6d8ac95a5dde312e0b7401b472cfb9/api/types/container/host_config.go#L308) 
where 
key is a mount point and value is a mount option string passed to the mount syscall (can be empty)

**Comment by gsf-greece** [comment]
I made some efforts to add it myself, but only managed to get as far as this exception
`Caused by: io.fabric8.maven.docker.access.hc.http.HttpRequestException: linux mounts: Invalid tmpfs option ["/var/lib/mysql"] (Internal Server Error: 500)`

**Comment by rhuss** [comment]
Thanks a lot for your efforts, I will try to have a look asap.

**Comment by gsf-greece** [comment]
Actually I made this work :) The exception was only because I failed to give the path as a key. I will leave it to you since I'm certainly missing something, but at least I know it can be done.

**Comment by rhuss** [comment]
I'm going to make a new release right now, and will look into this issue right after.

**Comment by rhuss** [comment]
@gsf-greece are the mount options important to you ? Otherwise I would set it to an empty string all the time.

**Comment by gsf-greece** [comment]
Hi again, and thanks for taking a look into this. I'm not really sure what mount options are required. I usually add size=1G because I can't mount it locally without defining a size for tmpfs.

I realise adding support for tmpfs is somewhat trivial, and the most difficult part is finding a good way to add the xml syntax. I think in the long run they should be added as a separate xml element. for example 

> <mount><options><size>1G</size></options></mount>

For now, maybe letting the user define the options would be a more simple approach, but I don't know if that will create any security issues or stability issues, in case wrong mount options are used.

I will attach the patch file I made on friday, which is rather ugly, but is what I used to successfuly run it on my system. (Empty options element that can be filled by the user)
[unnamed.txt](https://github.com/fabric8io/docker-maven-plugin/files/553640/unnamed.txt)
